packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -17,6 +17,17 @@ const MISSING_TRANSLATION = Symbol('Missing translation');
 const PLURALIZATION_KEY_NAME = 'count';
 const SEPARATOR = '.';
 
+const numberFormats = new Map();
+export const memoizedNumberFormatter = function(locale, options = {}) {
+  const key = numberFormatCacheKey(locale, options);
```

Comment:
because we are using this at at least two places now, I think we should just add an `enabledOnServer` option to `memoize`. But that can also be done in a follow up PR.